mediators/datamapper/components/org.wso2.carbon.gateway.mediators.datamapper.engine/src/main/java/org/wso2/carbon/gateway/mediators/datamapper/engine/core/executors/ScriptExecutor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.gateway.mediators.datamapper.engine.core.executors;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.exceptions.JSException;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.exceptions.SchemaException;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.mapper.JSFunction;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.mapper.MappingResource;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.models.MapModel;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.models.Model;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.core.models.StringModel;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.output.formatters.MapOutputFormatter;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.utils.DataMapperEngineConstants;
+import org.wso2.carbon.gateway.mediators.datamapper.engine.utils.DataMapperEngineUtils;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import static org.wso2.carbon.gateway.mediators.datamapper.engine.utils.DataMapperEngineConstants.EQUALS_SIGN;
+import static org.wso2.carbon.gateway.mediators.datamapper.engine.utils.DataMapperEngineConstants.PROPERTIES_OBJECT_NAME;
+
+import java.util.Map;
+
+/**
+ * This class implements script executor for data mapper using java script executor (Rhino or
+ * Nashorn)
+ */
+public class ScriptExecutor implements Executor {
+
+    private static final Log log = LogFactory.getLog(ScriptExecutor.class);
+    private ScriptEngine scriptEngine;
+
+    /**
+     * Create a script executor of the provided script executor type
+     *
+     * @param scriptExecutorType
+     */
+    public ScriptExecutor(ScriptExecutorType scriptExecutorType) {
+        switch (scriptExecutorType) {
+            case NASHORN:
+                scriptEngine = new ScriptEngineManager().getEngineByName(DataMapperEngineConstants.NASHORN_ENGINE_NAME);
+                log.debug("Setting Nashorn as Script Engine");
```

Comment:
mmm I don't think it will be a overhead, and anyway IMO it should be consistant, and remember there was some mail discussion around this as well, As I remember conclusion was to add this